src/reactviews/pages/QueryResult/queryResultPane.tsx

```diff
@@ -0,0 +1,241 @@
+import { useVscodeWebview } from '../../common/vscodeWebviewProvider';
+import SlickGrid, { SlickGridHandle } from "./slickgrid";
+
+const useStyles = makeStyles({
+	root: {
+		width: '100%',
+		height: '100%',
+		display: 'flex',
+		flexDirection: 'column',
+	},
+	ribbon: {
+		width: '100%',
+		display: 'flex',
+		flexDirection: 'row',
+		'> *': {
+			marginRight: '10px'
+		},
+	},
+	queryResultPaneTabs: {
+		flex: 1,
+	},
+	tabContent: {
+		...shorthands.flex(1),
+		width: '100%',
+		height: '100%',
+		display: 'flex',
+		...shorthands.overflow('auto'),
+	},
+	queryResultContainer: {
+		width: '100%',
+		height: '100%',
+		position: 'relative',
+	},
+	queryResultPaneOpenButton: {
+		position: 'absolute',
+		top: '0px',
+		right: '0px',
+	},
+	messagesContainer: {
+		width: '100%',
+		height: '100%',
+		flexDirection: 'column',
+		'> *': {
+			marginBottom: '10px'
+		},
+
+	},
+	messagesRows: {
+		flexDirection: 'row',
+		...shorthands.padding('10px'),
+		'> *': {
+			marginRight: '10px'
+		},
+	}
+});
+
+export const QueryResultPane = () => {
+	const classes = useStyles();
+	const state = useContext(QueryResultContext);
+	const webViewState = useVscodeWebview<qr.QueryResultWebviewState, qr.QueryResultReducers>();
+	webViewState;
+	var metadata = state?.state;
+
+	const getVscodeTheme = (theme: Theme) => {
+
+		switch (theme) {
+			case webDarkTheme:
+				return 'vs-dark';
+			case teamsHighContrastTheme:
+				return 'hc-black';
+			default:
+				return 'light';
+		}
+	};
+
+	getVscodeTheme;
+
+	const columnsDef: TableColumnDefinition<qr.IMessage>[] = [
+		createTableColumn({
+			columnId: 'time',
+			renderHeaderCell: () => <>Timestamp</>
+		}),
+		createTableColumn({
+			columnId: 'message',
+			renderHeaderCell: () => <>Message</>
+		}),
+	];
+	const [columns] = useState<TableColumnDefinition<qr.IMessage>[]>(columnsDef);
+	const items = metadata?.messages ?? [];
+
+	const sizingOptions: TableColumnSizingOptions = {
+		'time': {
+			minWidth: 50,
+			idealWidth: 50,
+			defaultWidth: 50
+		},
+		'message': {
+			minWidth: 500,
+			idealWidth: 500,
+			defaultWidth: 500
+		}
+	};
+
+	const [columnSizingOption] = useState<TableColumnSizingOptions>(sizingOptions);
+	const { getRows, columnSizing_unstable, tableRef } = useTableFeatures(
+		{
+			columns,
+			items: items
+		},
+		[useTableColumnSizing_unstable({ columnSizingOptions: columnSizingOption })]
+	);
+	const rows = getRows();
+
+	if (!metadata) {
+		return null;
+	}
+
+	const gridRef = useRef<SlickGridHandle>(null);
+
+	return <div className={classes.root}>
+		<div className={classes.ribbon}>
+			<TabList
+				size='medium'
+				selectedValue={metadata.tabStates!.resultPaneTab}
+				onTabSelect={(_event, data) => {
+					state?.provider.setResultTab(data.value as qr.QueryResultPaneTabs);
+				}}
+				className={classes.queryResultPaneTabs}
+			>
+				{metadata.resultSetSummary && <Tab value={qr.QueryResultPaneTabs.Results} key={qr.QueryResultPaneTabs.Results}>
+					Results
```

Comment:
Please localize the strings in a follow up PR.

Reply:
updated